aiokafka/coordinator/assignors/abstract.py

```diff
@@ -1,20 +1,33 @@
 import abc
 import logging
+from typing import Dict, Set
+
+from aiokafka.cluster import ClusterMetadata
+from aiokafka.coordinator.protocol import (
+    ConsumerProtocolMemberAssignment,
+    ConsumerProtocolMemberMetadata,
+)
 
 log = logging.getLogger(__name__)
 
 
-class AbstractPartitionAssignor:
+class AbstractPartitionAssignor(abc.ABC):
     """Abstract assignor implementation which does some common grunt work (in particular
     collecting partition counts which are always needed in assignors).
     """
 
-    @abc.abstractproperty
-    def name(self):
+    @property
+    @abc.abstractmethod
+    def name(self) -> str:
         """.name should be a string identifying the assignor"""
 
+    @classmethod
     @abc.abstractmethod
-    def assign(self, cluster, members):
+    def assign(
+        cls,
+        cluster: ClusterMetadata,
+        members: Dict[str, ConsumerProtocolMemberMetadata],
```

Comment:
But it's like `return type` from https://typing.readthedocs.io/en/latest/source/best_practices.html#arguments-and-return-types (because `GroupCoordinator` always call this method with dict). But I watch only actual types, if this does not make sense, I will change this (and below) to protocols.

Reply:
I don't get it. Why is it like return type, when it _is_ argument type? Anyway, it's a question, I'm not sure what will be better. It's abstract class, so those variants are not fully compatible.